mlserver/model_wrapper.py

```diff
@@ -0,0 +1,78 @@
+from typing import Optional, List, Any
+
+from mlserver.codecs import InputCodecLike, RequestCodecLike
+
+from mlserver import MLModel, ModelSettings
+from mlserver.types import (
+    MetadataTensor,
+    RequestInput,
+    InferenceRequest,
+    MetadataModelResponse,
+    InferenceResponse,
+)
+
+
+class WrapperMLModel(MLModel):
```

Comment:
Do you remember why we needed this instead of just extending the `MLModel` class?

Reply:
It is essentially so that we have different implementations of `MLModel` where inner `MLModel` in this wrapper is created by a factory method and allowing the user to have in the `model-settings.json` one reference in `implementation` to this WrapperMLModel. For example in `mlserver-alibi-explain` runtime it is going to be `AlibiExplainRuntime` for ally types of alibi models.

Reply:
I simplified the code based on your suggestion to use `__new__` instead